### PR TITLE
Dn reliable websockets #7

### PR DESCRIPTION
**Pull request by dnhuch**
Add support for acknowledging events for reliable resources and create persistent/reconnectable subscriptions

### CODE REVIEW

File: src/main/java/io/vantiq/client/Vantiq.java [hunk 1]
```diff
@@ -1365,17 +1372,38 @@ public void subscribe(String resource,
...
+                               TypeOperation operation,
+                               SubscriptionCallback callback) {
+
+       subscribe(resource, id, operation, callback, null);
     }
 
     /**
-     * Unsubscribes to all current subscriptions by closing the WebSocket to the Vantiq
+     * Un
+     * s to all current subscriptions by closing the WebSocket to the Vantiq
```
Comment (rblumer4) [inline src/main/java/io/vantiq/client/Vantiq.java]: accidental typo here, I think.

File: src/main/java/io/vantiq/client/Vantiq.java [hunk 2]
```diff
@@ -1364,9 +1371,24 @@ public void subscribe(String resource,
...
-        this.session.subscribe(path, callback, this.enablePings);
+    public void subscribe(String resource,
+                          String id,
+                          TypeOperation operation,
+                          SubscriptionCallback callback) {
+
+        subscribe(resource, id, operation, callback, null);
     }
+    
+    /**Acknowledge the receipt of a reliable message*/
```
Comment (JakeSchmitz) [inline src/main/java/io/vantiq/client/Vantiq.java]: to be consistent with the other method comments this should be multiline:

```
/**
    Acknowledge the receipt of a reliable message
*/
```

File: src/main/java/io/vantiq/client/internal/VantiqSubscriber.java [hunk 3]
```diff
@@ -64,16 +64,36 @@ public void connect(VantiqSubscriberLifecycleListener lifecycleHandler) {
...
+            this.resourceId = path;
+            if (parameters != null) {
+                this.parameters.putAll(parameters);
+            }
+            this.parameters.put("requestId", path);
+
+            this.accessToken = accessToken;
+        }
+    } 
+    public void subscribe(String path, SubscriptionCallback callback, Map<String, Object> parameters) {
```
Comment (JakeSchmitz) [inline src/main/java/io/vantiq/client/internal/VantiqSubscriber.java]: newline between the end of the class def and function def please

File: src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java [hunk 4]
```diff
@@ -31,23 +33,25 @@
...
+    private static String token = null;
+    
     @BeforeClass
     public static void setUpIntgTest() throws Exception {
         // Pull values from java properties, so the credentials are not checked in
         server   = System.getProperty("server");
         username = System.getProperty("username");
         password = System.getProperty("password");
-
-        if(server == null || username == null || password == null) {
```
Comment (JakeSchmitz) [inline src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java]: Leave this check in, just extend it to also check the token

File: src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java [hunk 5]
```diff
@@ -31,23 +33,25 @@
...
+        token = System.getProperty("token");
     }
 
     @Before
     public void setUp() throws Exception {
         vantiq = new Vantiq(server);
-        vantiq.authenticate(username, password, handler);
+        if (!username.equals("") && !password.equals("")) {
+            vantiq.authenticate(username, password);
+        } else {
```
Comment (JakeSchmitz) [inline src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java]: This doesn't account for if the token wasn't specified

**Pull request merged**